src/parallel/HighsTaskExecutor.h

```diff
@@ -85,47 +90,75 @@ class HighsTaskExecutor {
     return nullptr;
   }
 
-  void run_worker(int workerId) {
-    // spin until the global executor pointer is set up
-    ExecutorHandle* executor;
-    // Following yields warning C4706: assignment within conditional
-    // expression when building libhighs on Windows (/W4):
-    //
-    //    while (!(executor = mainWorkerHandle.load(std::memory_order_acquire)))
-    //      HighsSpinMutex::yieldProcessor();
-    while (true) {
-      executor = mainWorkerHandle.load(std::memory_order_acquire);
-      if (executor != nullptr) break;
-      HighsSpinMutex::yieldProcessor();
-    }
-    // now acquire a reference count of the global executor
-    threadLocalExecutorHandle() = *executor;
-    HighsSplitDeque* localDeque = workerDeques[workerId].get();
-    threadLocalWorkerDeque() = localDeque;
-    HighsTask* currentTask = workerBunk->waitForNewTask(localDeque);
-    while (currentTask != nullptr) {
-      localDeque->runStolenTask(currentTask);
-
-      currentTask = random_steal_loop(localDeque);
-      if (currentTask != nullptr) continue;
-
-      currentTask = workerBunk->waitForNewTask(localDeque);
+  static void run_worker(int workerId, HighsTaskExecutor* ptr) {
+    auto& executorHandle = threadLocalExecutorHandle();
+    executorHandle.ptr = ptr;
+
+    // check if main thread has shutdown before thread has started
+    // if (ptr->mainWorkerId.load() != std::thread::id()) {
+    if (!(executorHandle.isMain)) {
```

Comment:
This should be a check to the shared `isStopped`, i.e., have we stopped before the thread has started.  The `isMain` check is not needed.  

I was using the `mainWorkerId == thread::id` to implicitly check if we had stopped - the explicit `isStopped` is much better!